### PR TITLE
(BOLT-752) Update Puppet for spec revision 3, fix for task type change

### PR DESCRIPTION
**Pull request by MikaelSmith**
To support revision 3 of the task specification, Puppet's representation of a Task type was changed to mirror Puppet Server's API: `name`, `files`, `metadata`. Add a more complete Task abstraction to Bolt - that's shared between Bolt and bolt-server - over Puppet's representation.

### CODE REVIEW

File: bolt-modules/boltlib/lib/puppet/functions/run_task.rb [hunk 1]
```diff
@@ -133,16 +135,16 @@ def run_task_raw(task_name, targets, description = nil, task_args = nil, &block)
...
-        if task.parameters[k] && task.parameters[k]['sensitive']
+        if params[k] && params[k]['sensitive']
           use_args[k] = Puppet::Pops::Types::PSensitiveType::Sensitive.new(v)
         end
       end
     end
 
     if executor.noop
-      if task.supports_noop
+      if task['metadata']['supports_noop']
```
Comment (adreyer) [inline bolt-modules/boltlib/lib/puppet/functions/run_task.rb]: Looking at this compared to the older method it seems like we'll need to spread default logic around a lot more than if "task" was a ruby object that we could add methods too.

Comment (MikaelSmith) [reply bolt-modules/boltlib/lib/puppet/functions/run_task.rb]: Yeah, I think that's the next thing I'm looking at cleaning up.

File: lib/bolt/outputter/human.rb [hunk 2]
```diff
@@ -119,9 +119,10 @@ def print_task_info(task)
         task_info = +""
         usage = +"bolt task run --nodes <node-name> #{task['name']}"
 
-        if task['parameters']
-          replace_data_type(task['parameters'])
-          task['parameters'].each do |k, v|
+        params = task['metadata']['parameters']
+        if params
+          replace_data_type(params)
```
[the remark this answers is not in the document]
Comment (MikaelSmith) [reply lib/bolt/outputter/human.rb]: Do we care about this in the JSON output as well?

[the remark this answers is not in the document]
Comment (MikaelSmith) [reply lib/bolt/outputter/human.rb]: i.e. if no type was explicitly specified, should we still put in the default `Any`, or leave it null?

[the remark this answers is not in the document]
Comment (adreyer) [reply lib/bolt/outputter/human.rb]: I don't. I'm not sure what the behavior should be there, probably just passing params through unmodified.

**Pull request merged**